src/Sidekick.Business/StartupExtensions.cs

```diff
@@ -28,21 +30,26 @@ public static IServiceCollection AddSidekickBusinessServices(this IServiceCollec
             services.AddSingleton<IHttpClientProvider, HttpClientProvider>();
             services.AddSingleton<IItemParser, ItemParser>();
             services.AddSingleton<ILanguageProvider, LanguageProvider>();
-            services.AddSingleton<IPoeDbClient, PoeDbClient>();
             services.AddSingleton<IPoeNinjaClient, PoeNinjaClient>();
             services.AddSingleton<IPoePriceInfoClient, PoePriceInfoClient>();
-            services.AddSingleton<IPoeWikiClient, PoeWikiClient>();
             services.AddSingleton<ITokenizer, ItemNameTokenizer>();
             services.AddSingleton<ITradeClient, TradeClient>();
             services.AddSingleton<IWhisperService, WhisperService>();
+            services.AddSingleton<IPoeDbClient, PoeDbClient>();
+            services.AddSingleton<IPoeWikiClient, PoeWikiClient>();
+
+            services.AddSingleton(serviceProvider =>
+                serviceProvider.GetService<SidekickSettings>().Wiki_Preferred == WikiSetting.PoeDb
+                    ? (IWikiProvider) serviceProvider.GetRequiredService<IPoeDbClient>()
+                    : serviceProvider.GetRequiredService<IPoeWikiClient>());
```

Comment:
Does this update when we change the settings or do we need to restart the application?

Reply:
Everything is registered as a singleton including the configuration so restart would be required. Another way would be to use the configuration `Options` and build a factory that returns the correct implementation based on current configuration.

Reply:
Yeah that is an issue. Once we have implemented the new view models for every view we will be able to register scoped services reliably. I will create a new issue to fix this.

References #220